### ISSUE TITLE
[Typescript] Error in typings 

### ISSUE THREAD

**Issue by jedrzejowski** [mikbry/material-ui-color#110]
**Describe the bug**
It seams there is a bug in typings which prevents from successful compilation.

**To Reproduce**
I have created a test repo
```
git clone https://github.com/jedrzejowski/my-issue-1
cd my-issue-1/
npm install
npm run tsc
```

**Expected behavior**
Code compiles

**Screenshots**
![image](https://user-images.githubusercontent.com/15786714/109289201-27fe4e00-7826-11eb-80d9-6076cca73f13.png)

`T` extends different types here: 
https://github.com/mikbry/material-ui-color/blob/99b673cc962f870ac3d0c38fdc9c66e5448cb4a5/src/types/index.d.ts#L57
and here:
https://github.com/mikbry/material-ui-color/blob/99b673cc962f870ac3d0c38fdc9c66e5448cb4a5/src/types/index.d.ts#L61

**Comment by mikbry** [comment]
In your tsconfig you are using `"strictNullChecks": true`removing it will compile ok.
I will fix it in next release

**Comment by mikbry** [comment]
fixed in v1.0.0
